### PR TITLE
PARQUET-1855: [C++] Improve parquet *MetaData documentation

### PR DESCRIPTION
**Pull request by fsaintjacques**

### CODE REVIEW

File: cpp/src/parquet/metadata.h [hunk 1]
```diff
@@ -178,27 +179,42 @@ class PARQUET_EXPORT ColumnChunkMetaData {
...
+  /// \brief Return the ColumnChunkMetaData of the corresponding column ordinal.
+  ///
+  /// WARNING, the returned object references memory location in it's parent
+  /// (RowGroupMetaData) object. Hence, the parent must outlive the returned
+  /// object.
+  ///
+  /// \param[i] index of the ColumnChunkMetaData to retrieve.
+  ///
+  /// \throws ParquetException if the index is out of bound.
+  std::unique_ptr<ColumnChunkMetaData> ColumnChunk(int i) const;
```
Comment (fsaintjacques) [inline cpp/src/parquet/metadata.h:206]: @wesm I believe we should make a breaking change and return either a const-ref or const-ptr (like SchemaDescriptor) and hold the corresponding ColumnChunks in a private vector.

Comment (wesm) [reply cpp/src/parquet/metadata.h:206]: Doesn't sound like a bad idea, can you open an issue about this?

**Review by wesm — APPROVED** [review]
+1, modulo CI

File: cpp/src/parquet/metadata.h [hunk 2]
```diff
@@ -178,27 +179,42 @@ class PARQUET_EXPORT ColumnChunkMetaData {
...
+  /// parent's column ordering.
   int num_columns() const;
+
+  /// \brief Return the ColumnChunkMetaData of the corresponding column ordinal.
+  ///
+  /// WARNING, the returned object references memory location in it's parent
+  /// (RowGroupMetaData) object. Hence, the parent must outlive the returned
+  /// object.
+  ///
+  /// \param[i] index of the ColumnChunkMetaData to retrieve.
```
Comment (wesm) [inline cpp/src/parquet/metadata.h:203]: doxygen syntax

File: cpp/src/parquet/metadata.h [hunk 3]
```diff
@@ -211,41 +227,72 @@ class PARQUET_EXPORT RowGroupMetaData {
...
 
+  /// \brief The number of row groups in the file.
   int num_row_groups() const;
+
+  /// \brief Return the RowGroupMetaData of the corresponding row group ordinal.
+  ///
+  /// WARNING, the returned object references memory location in it's parent
+  /// (FileMetaData) object. Hence, the parent must outlive the returned object.
+  ///
+  /// \param[i] index of the RowGroup to retrieve.
```
Comment (wesm) [inline cpp/src/parquet/metadata.h:265]: doxygen syntax

**Comment by fsaintjacques** [comment]
Spurious s3 failure.

**Comment by fsaintjacques** [comment]
@wesm ready to merge.

**Pull request closed without merging**